api/datastore/postgres/postgres.go

```diff
@@ -411,15 +401,18 @@ func scanRoute(scanner rowScanner, route *models.Route) error {
 		&headerStr,
 		&configStr,
 	)
+	if err != nil {
+		return err
+	}
 
 	if headerStr == "" {
 		return models.ErrRoutesNotFound
 	}
```

Comment:
Additionally, this `headerStr` check was happening *before* checking the `Scan` err.